src/test/java/de/tum/in/www1/artemis/exam/ExamIntegrationTest.java

```diff
@@ -323,54 +318,31 @@ void testRemoveStudentToExam_testExam() throws Exception {
     @Test
     @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
     void testRegisterUsersInExam() throws Exception {
-        userUtilService.addStudents(TEST_PREFIX, 5, 10);
         jiraRequestMockProvider.enableMockingOfRequests();
 
-        var exam = ExamFactory.generateExam(course1);
-        var savedExam = examRepository.save(exam);
-        examUtilService.addExamChannel(savedExam, "test exam");
-        var student1 = userUtilService.getUserByLogin(TEST_PREFIX + "student1");
-        var student2 = userUtilService.getUserByLogin(TEST_PREFIX + "student2");
-        var student3 = userUtilService.getUserByLogin(TEST_PREFIX + "student3");
-        var student5 = userUtilService.getUserByLogin(TEST_PREFIX + "student5");
-        var student6 = userUtilService.getUserByLogin(TEST_PREFIX + "student6");
-        var student7 = userUtilService.getUserByLogin(TEST_PREFIX + "student7");
-        var student8 = userUtilService.getUserByLogin(TEST_PREFIX + "student8");
-        var student9 = userUtilService.getUserByLogin(TEST_PREFIX + "student9");
-        var student10 = userUtilService.getUserByLogin(TEST_PREFIX + "student10");
-        var registrationNumber1 = "1111111";
-        var registrationNumber2 = "1111112";
-        var registrationNumber3 = "1111113";
-        var registrationNumber3WithTypo = registrationNumber3 + "0";
-        var registrationNumber5 = "1111115";
-        var registrationNumber5WithTypo = registrationNumber5 + "1";
-        var registrationNumber6 = "1111116";
+        var savedExam = examUtilService.addExam(course1);
+
+        List<String> registrationNumbers = Arrays.asList("1111111", "1111112", "1111113");
+        List<User> students = new ArrayList<>();
+
+        for (int i = 1; i < 4; i++) {
+            students.add(userUtilService.setRegistrationNumberOfUserAndSave(TEST_PREFIX + "student" + i, registrationNumbers.get(i - 1)));
+        }
```

Comment:
Could this be moved into a new `UserUtilService` function that sets the registration number of multiple users?

Reply:
This makes sense, thanks for the suggestion!